x/auth/account.go

```diff
@@ -115,6 +118,133 @@ func (acc *BaseAccount) SetSequence(seq int64) error {
 	return nil
 }
 
+// VestingAccount is an account that can define a vesting schedule
+// Vesting coins can still be delegated, but only transferred after they have vested
+type VestingAccount interface {
+	Account
+
+	// Returns true if account is still vesting, else false
+	// CONTRACT: After account is done vesting, account behaves exactly like BaseAccount
+    IsVesting(time.Time) bool
+
+    // Calculates amount of coins that can be sent to other accounts given the current time
+	SendableCoins(time.Time) sdk.Coins
+	// Called on bank transfer functions (e.g. bank.SendCoins and bank.InputOutputCoins)
+	// Used to track coins that are transferred in and out of vesting account after initialization
+	TrackTransfers(sdk.Coins)
+}
+
+// Implement Vesting Interface. Continuously vests coins linearly from StartTime until EndTime
+type ContinuousVestingAccount struct {
+	BaseAccount
+    OriginalVestingCoins sdk.Coins // Coins in account on Initialization
+    TransferredCoins     sdk.Coins // Net coins transferred into and out of account. May be negative
+
+    // StartTime and EndTime used to calculate how much of OriginalCoins is unlocked at any given point
+    StartTime time.Time
+    EndTime   time.Time
+}
+
+func NewContinuousVestingAccount(addr sdk.AccAddress, originalCoins sdk.Coins, startTime, endTime time.Time) ContinuousVestingAccount {
+	bacc := BaseAccount{
+		Address: addr,
+		Coins: originalCoins,
+	}
+	return ContinuousVestingAccount{
+		BaseAccount: bacc,
+		OriginalVestingCoins: originalCoins,
+		StartTime: startTime,
+		EndTime: endTime,
+	}
+}
+
+// Implements VestingAccount interface.
+func (vacc ContinuousVestingAccount) IsVesting(blockTime time.Time) bool {
+	return blockTime.Unix() > vacc.EndTime.Unix()
+}
+
+// Implement Vesting Account interface. Uses time in context to calculate how many coins
+// has been released by vesting schedule and then accounts for unlocked coins that have
+// already been transferred or delegated
+func (vacc ContinuousVestingAccount) SendableCoins(blockTime time.Time) sdk.Coins {
+	unlockedCoins := vacc.TransferredCoins
+	scale := float64(blockTime.Unix() - vacc.StartTime.Unix()) / float64(vacc.EndTime.Unix() - vacc.StartTime.Unix())
+
+	// Add original coins unlocked by vesting schedule
+	for _, c := range vacc.OriginalVestingCoins {
```

Comment:
I don't understand why we need this loop - can't we just subtract `TransferredCoins` from `OriginalVestingCoins`?